src/envoy/http/backend_routing/filter_fuzz_test.cc

```diff
@@ -22,46 +22,43 @@ DEFINE_PROTO_FUZZER(
   try {
     TestUtility::validate(input);
 
-    // This fuzz test only requires a single backend routing rule.
-    // All other rules are ignored. So improve performance by only allowing
-    // configs with one rule through.
-    if (input.config().rules_size() != 1) {
-      throw ProtoValidationException("Only 1 backend rule is allowed", input);
+    if (input.config().rules_size() < 1) {
+      throw ProtoValidationException("At least 1 backend rule needed", input);
     }
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "Controlled proto validation failure: {}", e.what());
+    return;
+  }
 
-    // Setup mocks.
-    NiceMock<Envoy::Http::MockStreamDecoderFilterCallbacks>
-        mock_decoder_callbacks;
-    NiceMock<Envoy::Server::Configuration::MockFactoryContext>
-        mock_factory_context;
-
-    // Set the operation name using the first backend routing rule.
-    Utils::setStringFilterState(
-        *mock_decoder_callbacks.stream_info_.filter_state_, Utils::kOperation,
-        input.config().rules(0).operation());
+  // Setup mocks.
+  NiceMock<Envoy::Http::MockStreamDecoderFilterCallbacks>
+      mock_decoder_callbacks;
+  NiceMock<Envoy::Server::Configuration::MockFactoryContext>
+      mock_factory_context;
 
-    // Set the variable binding query params.
-    Utils::setStringFilterState(
-        *mock_decoder_callbacks.stream_info_.filter_state_, Utils::kQueryParams,
-        input.binding_query_params());
+  // Set the operation name using the first backend routing rule.
+  Utils::setStringFilterState(
+      *mock_decoder_callbacks.stream_info_.filter_state_, Utils::kOperation,
+      input.config().rules(0).operation());
 
-    // Create the filter.
-    FilterConfigSharedPtr config = std::make_shared<FilterConfig>(
-        input.config(), "fuzz-test-stats", mock_factory_context);
-    Filter filter(config);
-    filter.setDecoderFilterCallbacks(mock_decoder_callbacks);
+  // Set the variable binding query params.
+  Utils::setStringFilterState(
+      *mock_decoder_callbacks.stream_info_.filter_state_, Utils::kQueryParams,
+      input.binding_query_params());
 
-    // Generate the user request.
-    auto headers =
-        Envoy::Fuzz::fromHeaders<Envoy::Http::TestRequestHeaderMapImpl>(
-            input.user_request().headers());
+  // Create the filter.
+  FilterConfigSharedPtr config = std::make_shared<FilterConfig>(
```

Comment:
Normally, FilterConfig can throw exception,  so this can be run inside try-catch block.

Reply:
Done